#! /bin/sh /usr/share/dpatch/dpatch-run
## 07_luksAddKey.dpatch by Jonas Meurer <jonas@freesources.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: do only ask once for passphrase of the existing key in luksAddKey

@DPATCH@

--- cryptsetup-1.0.3/lib/setup.c
+++ cryptsetup-1.0.3/lib/setup.c
@@ -680,7 +680,7 @@
 	const char *device = options->device;
 	struct crypt_options optionsCheck = { 
 		.key_file = options->key_file,
-		.flags = options->flags & ~CRYPT_FLAG_VERIFY,
+		.flags = options->flags & ~CRYPT_FLAG_VERIFY_IF_POSSIBLE,
 	};
 	struct crypt_options optionsSet = { 
 		.key_file = options->new_key_file,
